lms/services/organization.py

```diff
@@ -36,6 +38,41 @@ def get_by_linked_guid(self, guid) -> List[Organization]:
             .all()
         )
 
+    def get_by_public_id(self, public_id: str):
```

Comment:
This is a long and boring method that could be moved elsewhere in the future (PublicIDService?) but I  reckon it's fine here for now.

Reply:
Maybe pull this out as a separate `_parse_public_id()` method to more cleanly separate the org'y stuff from the id related stuff?

I'm imagining it would be something like:

```python
public_id = self._parse_public_id(public_id, type_="org")
```

So you could assert what you were expecting.

Reply:
Looking ahead one step, could this be the first arg in a `search()` method? Otherwise I suspect we'll end-up having some duplication.

I'm thinking something like:

```python
def search(self, public_id=None):
    query =  self._db_session.query(Organization)

    if public_id:
        query.filter_by(_public_id=self._parse_public_id(public_id))

    return query.all()
```